src/main/java/org/opensearch/knn/plugin/BlockedModelSet.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.plugin;
```

Comment:
I would move this to the indices folder for now. The indices folder should probably be renamed to models but that for another review.